metadata-ingestion/src/datahub/ingestion/source/kafka/kafka_config.py

```diff
@@ -1,19 +1,83 @@
 from datahub.ingestion.source.state.stale_entity_removal_handler import (
     StatefulStaleMetadataRemovalConfig,
 )
 from datahub.ingestion.source.state.stateful_ingestion_base import (
     StatefulIngestionConfigBase,
 )
+from datahub.ingestion.source_config.operation_config import is_profiling_enabled
+
+
+class SchemalessFallback(ConfigModel):
+    """Configuration for schema-less fallback."""
+
+    enabled: bool = Field(
+        default=False,
+        description="Enable automatic fallback to schema-less processing for topics not found in the schema registry.",
+    )
+
+    sample_timeout_seconds: float = Field(
+        default=2.0,
+        description="Maximum time to spend sampling messages from a single topic (in seconds).",
+    )
+    sample_strategy: str = Field(
+        default="hybrid",
+        description="Sampling strategy: 'earliest' (scan from beginning), 'latest' (recent messages only), or 'hybrid' (try latest first, fallback to earliest).",
+    )
+
+
+class ProfilerConfig(GEProfilingConfig):
+    """
+    Kafka-specific profiling configuration that extends GEProfilingConfig.
+
+    Inherits ALL standard profiling functionality from GEProfilingConfig including:
+    - operation_config.lower_freq_profile_enabled: Whether to do profiling at lower frequency
+    - operation_config.profile_day_of_week: Day of week to run profiling (0=Monday, 6=Sunday)
+    - operation_config.profile_date_of_month: Date of month to run profiling (1-31)
+    - All include_field_* flags (null_count, distinct_count, min/max/mean/median/stddev, etc.)
+    - turn_off_expensive_profiling_metrics, field_sample_values_limit, max_number_of_fields_to_profile
+    - report_dropped_profiles, catch_exceptions, tags_to_ignore_sampling
+    - profile_nested_fields (useful for complex JSON/Avro), query_combiner_enabled
+
+    Additional Kafka-specific profiling options:
+    """
+
+    max_sample_time_seconds: PositiveInt = Field(
+        default=60,
+        description="Maximum time to spend sampling messages in seconds",
+    )
+    sampling_strategy: str = Field(
+        default="latest",
+        description="Strategy for sampling messages: 'latest' (from end of topic), 'random' (random offsets), 'stratified' (evenly distributed), 'full' (entire topic, respects sample_size)",
+    )
+    batch_size: PositiveInt = Field(
+        default=100,
+        description="Number of messages to fetch in a single batch (for more efficient reading)",
+    )
+
+    # Override sample_size from base class with Kafka-appropriate default
+    sample_size: int = Field(
+        default=1000,
+        description="Number of messages to sample for profiling. Higher values provide more accurate statistics but take longer to process.",
+    )
+
+    # Inherits max_workers from GEProfilingConfig - controls profiling and schema inference parallelization
+
+    # Kafka-specific field for handling complex nested JSON/Avro structures
+    flatten_max_depth: int = Field(
```

Comment:
we have the global `profile_nested_fields: bool`

how is recursion has never been an issue before and now it is?

alternatively, if recursion is a general concern:
* should this be promoted as general profile config? 
* or should preventing recursion errors strategically solved?